src/LuaPrinter.ts

```diff
@@ -117,132 +161,222 @@ export class LuaPrinter {
         }
     }
 
-    private printDoStatement(statement: tstl.DoStatement): string {
-        let result = this.indent("do\n");
+    private printDoStatement(statement: tstl.DoStatement): SourceNode {
+        const chunks: Array<(string | SourceNode)> = [];
+        chunks.push(this.indent("do\n"));
         this.pushIndent();
-        result += this.ignoreDeadStatements(statement.statements).map(s => this.printStatement(s)).join("");
+        chunks.push(...this.ignoreDeadStatements(statement.statements).map(s => this.printStatement(s)));
         this.popIndent();
-        result += this.indent("end\n");
+        chunks.push(this.indent("end\n"));
 
-        return result;
+        return this.createSourceNode(statement.line, statement.column, this.sourceFile, chunks);
     }
 
-    private printVariableDeclarationStatement(statement: tstl.VariableDeclarationStatement): string {
-        const left = this.indent(`local ${statement.left.map(e => this.printExpression(e)).join(", ")}`);
+    private printVariableDeclarationStatement(statement: tstl.VariableDeclarationStatement): SourceNode {
+        const chunks: Array<(string | SourceNode)> = [];
+        chunks.push("local ");
+        statement.left.forEach((e, i) => {
```

Comment:
Instead of repeating this each time, we should just make a `function join(elements: Array<string | SourceNode>, separator: string): Array<string | SourceNode>` to do this for us. Would also greatly clean this up.